app/models/section.rb

```diff
@@ -159,14 +160,18 @@ def eql?(other)
     id.eql?(other.id)
   end
 
-protected
-
-  attr_reader :slug_generator, :edition_factory
-
   def never_published?
     !published?
   end
 
+  def change_note_required?
+    !(never_published? || minor_update?)
```

Comment:
Or even `published? && major_update?` although it's not obvious that this alone would warrant the addition of a `major_update?` method.